### PR TITLE
Improve gradle performance significantly

### LINKED ISSUE #142: Gradle performance could be improved.
## Describe the issue

The gradle performance is currently sub-par and could be improved.

### PR DESCRIPTION
**Pull request by solonovamax**
# Pull Request

## Brief description.
The current gradle performance is sub-par and could be improved easily by allowing for multithreading.

### What Issues Does This Fix?

Fix #142

## Licensing

- [x] I am the original author of this code, and I am willing to release it under [GPLv3](https://www.gnu.org/licenses/gpl-3.0.en.html).
- [ ] I am not the original author of this code, but it is in public domain or released under [GPLv3](https://www.gnu.org/licenses/gpl-3.0.en.html) or a compatible license.
    

## Goal of the PR

- [x] Enable the new VFS watch feature in gradle 6.8.3 to allow the gradle daemon to act more intelligently when choosing what to recompile
- [x] Enable parallel execution of gradle tasks.
- [x] Give the gradle runner vm 2G of memory
- [x] Make the `:common` subproject no longer rely on the shadowJar plugin, as it is unneeded for that.

#### Types of changes

- [ ] Bug fix 
- [ ] New feature 
- [x] Improvement 
- [ ] Documentation change 
- [ ] Translation 

#### Compatiblity

- [ ] Breaking change 
- [x] Non-Breaking change.
    

#### Contribution Guidelines.

- [x] I have read the [`CONTRIBUTING.md`](https://github.com/PolyhedralDev/Terra/blob/master/CONTRIBUTING.md) document in the root of the git repository.
- [x] My code follows the code style for this project. 

#### Documentation

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

#### Testing

- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -5,6 +5,23 @@ val versionObj = Version("5", "2", "0", true)
...
+        options.isFork = true
+        options.isIncremental = true
+    }
+    tasks.withType<Test>().configureEach {
+        useJUnitPlatform()
+
+        maxHeapSize = "2G"
+        ignoreFailures = false
+        failFast = true
+        maxParallelForks = (Runtime.getRuntime().availableProcessors()).takeIf { it > 0 } ?: 1
```
Comment (dfsek) [inline build.gradle.kts:19]: This might be better as `availableProcessors()-1`

**Review by dfsek — CHANGES_REQUESTED** [review]

File: gradle.properties [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Magic options for more perf
+org.gradle.jvmargs=-Xmx2048M -XX:MaxMetaspaceSize=512m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8 -XX:+UseParallelGC
```
Comment (dfsek) [inline gradle.properties:2]: Have you tested this with Loom? We *might* want to bump this to 3 or even 4GB. Though I do believe newer Loom versions use significantly less memory. Other than that these changes LGTM

**Review by dfsek — APPROVED** [review]

File: gradle.properties (hunk 2 above)
Comment (solonovamax) [reply gradle.properties:2]: Nope! I'm going to test that out real quick, but if it _does_ fail, then I want to find a way to give loom 4gb without giving the _rest_ of the gradle daemons 4gb.

**Comment by solonovamax** [comment]
Those commits *should* fix any issues that were happening.

Also, we could probably go down to 1GB because loom actually launches a separate JVM process for the remapping and shit. (It's launched with 3G.)

**Pull request merged**